lib/activerecord_spanner_adapter/connection.rb

```diff
@@ -44,11 +44,20 @@ def self.information_schema config
     end
 
     def session
+      @last_used = Time.current
       @session ||= spanner.create_session instance_id, database_id
     end
     alias connect! session
 
     def active?
+      # This method should not initialize a session.
```

Comment:
The `active?` method itself should not make the connection active by (implicitly) creating a session.